### DISCUSSION TITLE
Calibration for valetudo?

### DISCUSSION THREAD

**Discussion by Raabi91** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#571] [Q&A]
Have today my dreame d9 with valetudo set up so far that it runs room cleaning. How do I now best caliber my cards so that I can also use zone cleaning. Unfortunately nothing found

**Comment by ksga** [comment]
It seems Hypfer isn't willing to publish coordinates, so I simply used trial and error. Setup a mqtt publish string looking at the example here: https://valetudo.cloud/pages/integrations/mqtt.html#zonecleaningzonecleaningcapability
Try increasing the zone with a factor 10 (for me it was positive values for x and y) until you see the zone being drawn on the map from [ICBINV](https://github.com/Hypfer/ICantBelieveItsNotValetudo) or on the Valetudo webserver. When you see something (it turned out 0,0 was somewhere roughly 15 meters outside the house), keep tweaking the values until all four corners are visible, save the map as a jpg and look up the image coordinates using a picture editor (GIMP or MS Paint works fine). Add the coordinate set to your card like this:
```c
calibration_source:
  calibration_points:
    - vacuum:
        x: 1520
        'y': 2000
      map:
        x: 8
        'y': 6
    - vacuum:
        x: 2300
        'y': 2000
      map:
        x: 321
        'y': 6
    - vacuum:
        x: 1520
        'y': 2690
      map:
        x: 8
        'y': 283
```
